letsencrypt/client/display/ops.py

```diff
@@ -6,41 +7,103 @@
 from letsencrypt.client import interfaces
 from letsencrypt.client.display import util as display_util
 
+
 # Define a helper function to avoid verbose code
 util = zope.component.getUtility  # pylint: disable=invalid-name
 
 
-def choose_authenticator(auths, errs):
-    """Allow the user to choose their authenticator.
+def choose_plugin(prepared, question):
+    """Allow the user to choose ther plugin.
 
-    :param list auths: Where each of type
-        :class:`letsencrypt.client.interfaces.IAuthenticator` object
-    :param dict errs: Mapping IAuthenticator objects to error messages
+    :param list prepared: List of `~.PluginEntryPoint`.
+    :param str question: Question to be presented to the user.
 
-    :returns: Authenticator selected
-    :rtype: :class:`letsencrypt.client.interfaces.IAuthenticator` or `None`
+    :returns: Plugin entry point chosen by the user.
+    :rtype: `~.PluginEntryPoint`
 
     """
-    descs = [auth.description if auth not in errs
-             else "%s (Misconfigured)" % auth.description
-             for auth in auths]
+    opts = [plugin_ep.name_with_description
```

Comment:
name_with_description doesn't display well.  I would recommend only including the description. (I realize the descriptions could technically clash... though I don't see this being a problem on our target audience's systems)
```
How would you like to authenticate with Let's Encrypt CA?
-------------------------------------------------------------------------------
1: apache (Apache Web Server)
2: standalone (Standalone Authenticator)
-------------------------------------------------------------------------------
```